easybuild/easyblocks/b/blender.py

```diff
@@ -0,0 +1,101 @@
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Blender, implemented as an easyblock
+
+@author: Samuel Moors (Vrije Universiteit Brussel)
+"""
+import glob
+import os
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Blender(CMakeMake):
+    """Support for building Blender."""
+
+    def configure_step(self):
+        """Set CMake options for Blender"""
+        self.cfg['separate_build_dir'] = True
+
+        default_config_opts = {
+            'WITH_INSTALL_PORTABLE': 'OFF',
+            'WITH_BUILDINFO': 'OFF',
+            # disable SSE detection to give EasyBuild full control over optimization compiler flags being used
+            'WITH_CPU_SSE': 'OFF',
+            'CMAKE_C_FLAGS_RELEASE': '-DNDEBUG',
+            'CMAKE_CXX_FLAGS_RELEASE': '-DNDEBUG',
+            # these are needed unless extra dependencies are added for them to work
+            'WITH_GAMEENGINE': 'OFF',
+            'WITH_SYSTEM_GLEW': 'OFF',
+        }
+        for key in default_config_opts:
+            opt = '-D%s=' % key
+            if opt not in self.cfg['configopts']:
+                self.cfg.update('configopts', opt + default_config_opts[key])
+
+        # Python paths
+        python_root = get_software_root('Python')
+        if python_root:
+            shlib_ext = get_shared_lib_ext()
+            pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+            site_packages = os.path.join(python_root, 'lib', 'python%s' % pyshortver, 'site-packages')
+            numpy_root = glob.glob(os.path.join(site_packages, 'numpy-*-py%s-linux-x86_64.egg' % pyshortver))[0]
+            requests_root = glob.glob(os.path.join(site_packages, 'requests-*-py%s.egg' % pyshortver))[0]
+            python_lib = glob.glob(os.path.join(python_root, 'lib', 'libpython%s*.%s' % (pyshortver, shlib_ext)))[0]
+            python_include_dir = glob.glob(os.path.join(python_root, 'include', 'python%s*' % pyshortver))[0]
```

Comment:
@smoors You're blindly using the 1st value here, but what if there are multiple results returned (or none)? Please make sure you only have a single result, and fail otherwise, something like:

```python
glob_pattern = os.path.join(python_root, 'include', 'python%s*' % pyshortver)
res = glob.glob(glob_pattern)
if len(res) == 1:
    python_include_dir = res[0]
else:
    raise EasyBuildError("Was expected exactly one match for '%s', found %d: %s", glob_pattern, len(res), res)
```

Ideally, introduce a function for this that takes the glob pattern and returns the result (or fails), since you need it in a bunch of places...

```python
python_include_dir = find_dir_for_glob_pattern(os.path.join(python_root, 'include', 'python%s*' % pyshortver))

Reply:
ok, I understand what you want now. I (perhaps wrongly) assumed it is reasonable to expect there cannot be more than 1 value, and if there is none it will fail anyway.
your suggestion is indeed safer and will give a more clear error.